src/pygeoprocessing/geoprocessing.py

```diff
@@ -3716,27 +3678,27 @@ def _gdal_to_numpy_type(band):
         numpy_datatype (numpy.dtype): equivalent of band.DataType
 
     """
-    # doesn't include GDT_Byte because that's a special case
-    base_gdal_type_to_numpy = {
-        gdal.GDT_Int16: numpy.int16,
-        gdal.GDT_Int32: numpy.int32,
-        gdal.GDT_UInt16: numpy.uint16,
-        gdal.GDT_UInt32: numpy.uint32,
-        gdal.GDT_Float32: numpy.float32,
-        gdal.GDT_Float64: numpy.float64,
-    }
+    if (GDAL_VERSION < (3, 7, 0) and gdal_type == gdal.GDT_Byte and
+        (('PIXELTYPE=SIGNEDBYTE' in metadata) or
+         ('PIXELTYPE' in metadata and metadata['PIXELTYPE'] == 'SIGNEDBYTE'))):
+            return numpy.int8
+
+    numpy_type = gdal_array.GDALTypeCodeToNumericTypeCode(gdal_type)
+    if numpy_type is None:
+        raise ValueError(f"Unsupported DataType: {gdal_type}")
+    return numpy_type
 
-    if band.DataType in base_gdal_type_to_numpy:
-        return base_gdal_type_to_numpy[band.DataType]
 
-    if band.DataType != gdal.GDT_Byte:
-        raise ValueError("Unsupported DataType: %s" % str(band.DataType))
+def _numpy_to_gdal_type(numpy_type):
```

Comment:
Could you add a docstring here that includes the input and output types?  Thanks!

Reply:
Added a docstring!